PPP.py

```diff
@@ -339,9 +364,27 @@ def getArguments():
     parser.add_argument('-retention', metavar='n', type=int, nargs=1, default=10,
                         help='Number of previous local playlist backups to keep (Default 10)')
 
+    parser.add_argument('-nocleanup', action='store_true',
+                        help='Disable removal of .tmp directory (for debug)')
+
     return parser.parse_args()
 
 
+def readPlaylistFile (filename):
+    _playlist = []
+    try:
+        with io.open(filename, "rt", encoding='utf8') as inf:
+            for line in inf:
+                if not line.startswith('#'):
+                    _playlist.append(line.strip())
+    except:
+        with io.open(filename, "rt", encoding='latin1') as inf:
```

Comment:
My only issue with this is that it assumes if io.open as utf8 fails, it assumes the encoding must be latin1

Is there a better way, that maybe recognises the encoding automatically?

Reply:
Fragile, I agree.  I'm not aware of being able to determine the encoding other than trying to open/read it.  The issue needs some research.